app/controllers/festivity_markets_controller.rb

```diff
@@ -0,0 +1,67 @@
+class FestivityMarketsController < ApplicationController
+  include Concerns::FestivityCustomPage
+  include Concerns::FestivitySearchCaching
+
+  def index
+
+    order_by = params[:sort] ? params[:sort] : "start_date"
+    @title = "#{current_site.festivity_festival_name}: Artist's Market"
+    @filter_type = current_site.festivity_filter_type
+    @markets =  Rails.cache.fetch("#{cache_key}", expires_in: 2.hours) do
+      FestivityMarketList.search(
+        {dates: search_dates.join(","),
+         categories: params[:categories],
+         filter_type: current_site.festivity_filter_type}, order_by
+      ).markets
+    end
+
+
+    @selected_dates = params[:dates] ? FestivityDatetimeFilterPresenter.parse(params[:dates].split(","), current_site.festivity_filter_type) : []
+    @selected_categories = params[:categories] ? params[:categories].split(",") : []
+    @selected_sort = order_by
+    # If the request is AJAX, only return the market list itself, not the full page
+    if request.xhr?
+      render partial: "market_list"
+    else
+      render 'index'
+    end
+
+  end
+
+  def show
+    @market = FestivityMarketPage.find_by_slug_and_status_id(params[:id], Status[:published].id)
+    if @market
+      @page = @market
+      @title = "#{current_site.festivity_festival_name}: #{@market.title}"
+    else
+      file_not_found_for_site
+    end
+
+  end
+
+  private
+
+  def cache_key
+    "markets-#{params[:categories.to_s]}-#{params[:dates].to_s}-#{params[:sort].to_s}-#{request.xhr?}-#{request.subdomain}.#{request.domain}"
```

Comment:
That happened to me with pagination on CD.org and the calendar filters. This makes sense.